### DISCUSSION TITLE
Simulation of solidification structure in additive manufacturing

### DISCUSSION THREAD

**Discussion by wowodejiajia** [idaholab/moose#19385] [Q&A General] [answered]
Dear MOOSE developers,

Can MOOSE simulate solidification structure simulation in additive manufacturing？If it can, please give an example.

**Comment by jiangwen84** [comment]
I guess you asked about the phase-field modeling of solidification. @SudiptaBiswas did you add any examples in MOOSE?

**Reply by SudiptaBiswas** [reply]
We have some solidification examples [here](https://github.com/idaholab/moose/tree/next/modules/phase_field/examples/anisotropic_interfaces).

**Reply by wowodejiajia** [reply]
Thank you.

**Comment by ngrilli** [comment, marked as answer]
Dear @wowodejiajia 

We have done this recent work:
https://arxiv.org/abs/2105.13257
which is now accepted in Computational Mechanics journal.

It is not based on phase field, but on elements moved from active to inactive subdomain and vice versa
to model solidification.

The simulation used in the paper is here:
https://github.com/ngrilli/c_pfor_am/tree/main/test/tests/Paper_ElementEliminationMethod

Nicolò

**Reply by wowodejiajia** [reply]
Thank you very much.
